tests/unit_tests/docmaps/v2/codecs/preprint_test.py

```diff
@@ -2,56 +2,75 @@
     get_docmap_preprint_input,
-    get_docmap_preprint_input_with_published_and_tdmpath
+    get_docmap_preprint_input_with_published_and_meca_path,
+    get_meca_path_content
 )
 from tests.unit_tests.docmaps.v2.test_data import (
     DOI_1,
     MANUSCRIPT_VERSION_1,
     PREPRINT_LINK_1,
     PREPRINT_VERSION_1,
-    TDM_PATH_1
+    MECA_PATH_1
 )
 
 
 class TestGetDocmapPreprintInput:
-    def test_should_populate_published_and_tdm_path_if_detailed_param_true(self):
-        result = get_docmap_preprint_input_with_published_and_tdmpath(
+    def test_should_populate_docmap_preprint_input(self):
+        result = get_docmap_preprint_input(
             manuscript_version=MANUSCRIPT_VERSION_1
         )
         assert result == {
             'type': 'preprint',
             'doi': DOI_1,
             'url': PREPRINT_LINK_1,
             'versionIdentifier': PREPRINT_VERSION_1,
-            'published': date.fromisoformat('2021-01-01').isoformat(),
-            '_tdmPath': TDM_PATH_1
         }
 
-    def test_should_return_empty_str_for_published_if_not_defined(self):
-        manuscript_version = {
-            **MANUSCRIPT_VERSION_1,
-            'preprint_published_at_date': None
+
+class TestGetMecaPathContent:
+    def test_should_populate_meca_path_content(self):
+        result = get_meca_path_content(MECA_PATH_1)
+        assert result == {
+            'type': 'web-manifestation',
+            'url': MECA_PATH_1
         }
-        result = get_docmap_preprint_input_with_published_and_tdmpath(
-            manuscript_version=manuscript_version
+
+
+class TestGetDocmapPreprintInputWithPublishedAndMecaPath:
+    def get_docmap_preprint_input_with_published_and_meca_path(self):
+        result = get_docmap_preprint_input(
+            manuscript_version=MANUSCRIPT_VERSION_1
         )
         assert result == {
             'type': 'preprint',
             'doi': DOI_1,
             'url': PREPRINT_LINK_1,
             'versionIdentifier': PREPRINT_VERSION_1,
-            'published': '',
-            '_tdmPath': TDM_PATH_1
+            'published': date.fromisoformat('2021-01-01').isoformat(),
+            'content': [
+                get_meca_path_content(MECA_PATH_1)
+            ]
         }
 
-    def test_should_not_populate_published_and_tdm_path_if_detailed_param_false(self):
-        result = get_docmap_preprint_input(manuscript_version=MANUSCRIPT_VERSION_1)
-        assert result == {
-            'type': 'preprint',
-            'doi': DOI_1,
-            'url': PREPRINT_LINK_1,
-            'versionIdentifier': PREPRINT_VERSION_1
+    def test_should_return_empty_str_for_published_if_not_defined(self):
+        manuscript_version = {
+            **MANUSCRIPT_VERSION_1,
+            'preprint_published_at_date': None
+        }
+        result = get_docmap_preprint_input_with_published_and_meca_path(
+            manuscript_version=manuscript_version
+        )
+        assert result['published'] == ''
```

Comment:
what happened with the empty date strings? I thought we came to the conclusion that they are not applicable. But otherwise they should be None or not present.

Reply:
I did not change this actually just shortened the test but I can update to None